mathesar/api/viewsets/tables.py

```diff
@@ -105,18 +107,21 @@ def previews(self, request, pk=None):
             raise ValidationError("Incorrect number of columns in request.")
 
         table_data = TableSerializer(table, context={"request": request}).data
-        try:
+        with exception_transformer({**dict.fromkeys([DataError, IntegrityError], ExceptionTransformerDetail(400,
+                                                                                                            4001,
```

Comment:
How do we generate numbers for error codes and make sure they are unique?

I think it might make sense to have a mapping of exceptions and error codes somewhere central and use it here.

Reply:
I will replace the error codes with Enum